### ISSUE TITLE
allMandatoryParameters and validationMethod = NONE don't work together

### ISSUE THREAD

**Issue by xtaixe** [immutables/immutables#1271]
Hi. It seems that there is an incompatibility between `allMandatoryParameters` and `validationMethod = NONE`.

When having something like:

```
@Value.Immutable
@Value.Style(
        allMandatoryParameters = true,
        typeAbstract = "_*",
        typeImmutable = "*",
        validationMethod = NONE,
        defaults = @Value.Immutable(builder = false))
public interface _MyType {

    String property();

}
```

only a no-args factory method `MyType.of()` is created instead of the expected `MyType.of(String property)`.

If I use `allParameters` instead of `allMandatoryParameters` or remove `validationMethod = NONE` everything works as expected.

**Comment by elucash** [comment]
I've reread `ValidationMethod.NONE` javadoc and it seems that it is making attributes non mandatory, and unfortunately enough users might rely on this behavior even if a little bit regrettable in a way it is named/documented. So "fixing" it is not an option.
I've introduced new `ValidationMethod.MANDATORY_ONLY` constant to still have non-nullable non-default attributes mandatory with the regular check for missing attributes, just omit null-check on value.

**Comment by elucash** [comment]
released in 2.9.0
